sbi/inference/snpe/snpe_base.py

```diff
@@ -162,10 +165,17 @@ def __call__(
 
             # Run simulations for the round.
             theta, x, prior_mask = self._run_simulations(round_, num_sims)
+
+            # Check for NaNs in simulations.
+            is_valid_x, num_nans, num_infs = handle_invalid_x(
+                x, exclude_invalid=self.handle_nans
+            )
+            warn_on_invalid_x(num_nans, num_infs, self.handle_nans)
```

Comment:
We could also put this at the end of `self._run_simulations(round_, num_sims, self.handle_nans)`. This would then directly return only the 'good' simulations.

It has the advantage of not having to call `handle_invalid_x` twice (second call for the embedding_net)

Reply:
good point, but the `_run_simulations` looks at the moment this would result in more code repetitions. I would leave at this level for now.

Reply:
Okey!